File: lesunny/027_정수제곱근판별.java
import java.lang.*;

class Solution {
    public long solution(long n) {
        long answer = 0;
        double temp = Math.sqrt(n);
        
        if(temp - (int)temp > 0)
            answer = -1;
        else
            answer = (long)Math.pow(temp+1,2);
        
        return answer;
    }
}

File: JIWON/027 정수 제곱근 판별.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    
    for(long long i=1;i<=n;i++){
        answer = -1;
        if(n == i*i){
            answer = (i+1)*(i+1);
            break;
        } 
    }
    
    return answer;
}
